apps/zui/src/views/session-page/handler.ts

```diff
@@ -66,16 +66,18 @@ export class SessionPageHandler extends ViewHandler {
     const history = this.select(Current.getHistory)
 
     fetchQueryInfo(program).then((info) => {
-      const {poolName, error} = info
-      const pool = this.select(Pools.getByName(lakeId, poolName))
-
+      if (info.error) {
+        return
+      }
       this.dispatch(QueryInfo.set({isParsed: true, ...info}))
+      const poolName = this.select(QueryInfo.getPoolName)
       this.invoke("updatePluginSessionOp", {poolName, program})
+      const pool = this.select(Pools.getByName(lakeId, poolName))
       if (pool && !pool.hasSpan()) {
         this.dispatch(syncPool(pool.id, lakeId))
       }
```

Comment:
Previously, all this stuff would happen regardless of the value of `info.error`. Why change that?

Reply:
Well, that certainly seems like the sort of thing I should have been able to figure out for myself. Maybe next time!